pilot/pkg/status/report.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"fmt"
+	"strings"
+
+	"istio.io/istio/pilot/pkg/config/kube/crd"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/config/schema/resource"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type DistributionReport struct {
+	Reporter            string
+	DataPlaneCount      int
+	InProgressResources map[string]int
+}
+
+func (r *DistributionReport) SetProgress(resource fmt.Stringer, progress int) {
+	r.InProgressResources[resource.String()] = progress
+}
+
+func ReportFromYaml(content []byte) (DistributionReport, error) {
+	out := DistributionReport{}
+	err := yaml.Unmarshal(content, &out)
```

Comment:
any reason to put this in the configmap like
```
data:
  distribution-report: |
    reporter: istiod-586cb9786b-wwztg
    dataplanecount: 12
    inprogressresources: {}
```

instead of
```
data:
    reporter: istiod-586cb9786b-wwztg
    dataplanecount: 12
    inprogressresources: {}
```
?

Reply:
I suspect that this would not work with the client-go Data field, which we set in reporter.go:217

Reply:
I'm even more confused. 

How do we report 'dataplanecount' ? There is a master election - so it's whatever one istiod replica generates.

Reply:
the reporter is part of the 'follower' code.  Each of the followers reports their individual counts to the leader, who aggregates and writes to status fields.